### PR TITLE
[Windows/ARM64] Fix raycast/embree ARM64 build with LLVM/MinGW.

### PR DESCRIPTION
**Pull request by bruvzg**

### CODE REVIEW

File: thirdparty/embree/common/simd/arm/sse2neon.h [hunk 1]
```diff
@@ -102,7 +102,9 @@
 #include <stdint.h>
 #include <stdlib.h>
 
-#if defined(_WIN32)
+// -- GODOT start --
+#if defined(_WIN32) && !defined(__clang__)
```
Comment (akien-mga) [inline thirdparty/embree/common/simd/arm/sse2neon.h:106]: Why doesn't it work on llvm-mingw specifically but works on mingw-w64-gcc?

File: thirdparty/embree/common/simd/arm/sse2neon.h [hunk 2]
```diff
@@ -2080,8 +2082,16 @@ FORCE_INLINE void *_mm_malloc(size_t size, size_t align)
...
+// -- GODOT start --
+#if defined(_WIN32)
+    ptr = _aligned_malloc(size, align);
+	if (ptr)
         return ptr;
+#else
+	if (!posix_memalign(&ptr, align, size))
+        return ptr;
+#endif
+// -- GODOT end --
```
Comment (akien-mga) [inline thirdparty/embree/common/simd/arm/sse2neon.h:2094]: There's mixed tabs and spaces indentation.

File: thirdparty/embree/common/simd/arm/sse2neon.h (hunk 1 above)
Comment (bruvzg) [reply thirdparty/embree/common/simd/arm/sse2neon.h:106]: Most likely it's not clang but UCRT specific, but `_mm_malloc` is not found with this define.

File: thirdparty/embree/common/simd/arm/sse2neon.h (hunk 2 above)
Comment (bruvzg) [reply thirdparty/embree/common/simd/arm/sse2neon.h]: Fixed.

File: thirdparty/embree/common/simd/arm/sse2neon.h (hunk 1 above)
Comment (bruvzg) [reply thirdparty/embree/common/simd/arm/sse2neon.h:106]: Also, this is ARM specific code, so MinGW-GCC might have it missing as well.

Comment (akien-mga) [reply thirdparty/embree/common/simd/arm/sse2neon.h:106]: Should we use `!defined(__MINGW32__)` then?

Comment (akien-mga) [reply thirdparty/embree/common/simd/arm/sse2neon.h:106]: Either way, worth PR'ing upstream to get more eyes on it and find out the proper fix there: https://github.com/DLTcollab/sse2neon

Downstream I'm fine with either option to get this to compile.

**Review by akien-mga — APPROVED** [review]

File: thirdparty/embree/common/simd/arm/sse2neon.h (hunk 1 above)
Comment (bruvzg) [reply thirdparty/embree/common/simd/arm/sse2neon.h]: MSVC define it for all architectures, and it's just a macro for the same `_aligned_malloc` I have added, so it's the same.

Comment (akien-mga) [reply thirdparty/embree/common/simd/arm/sse2neon.h]: Then maybe it should just be `_MSC_VER`?
Is it defined for clang-cl?

Comment (bruvzg) [reply thirdparty/embree/common/simd/arm/sse2neon.h]: No idea, but `__MINGW32__` definitely should not be defined.

Comment (bruvzg) [reply thirdparty/embree/common/simd/arm/sse2neon.h]: Upstream PR - https://github.com/DLTcollab/sse2neon/pull/637

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**